src/main/java/de/unistuttgart/iste/gits/reward/service/calculation/FitnessScoreCalculator.java

```diff
@@ -22,198 +23,164 @@ public class FitnessScoreCalculator implements ScoreCalculator {
+        }
 
-        double updatedFitnessScore = newFitnessScore + fitnessRegen * contentsToRepeat;
+        double updatedFitnessScore = fitnessScoreBefore.getValue() + fitnessRegen;
         int intUpdatedFitnessScore = (int) Math.round(updatedFitnessScore);
 
+        if (!isDueForRepetition(content)) {
+            // content was reviewed successfully but is not due for repetition
+            intUpdatedFitnessScore = fitnessScoreBefore.getValue() + 1;
+        }
+
         RewardScoreLogEntry logEntry = RewardScoreLogEntry.builder()
                 .date(OffsetDateTime.now())
-                .difference(intUpdatedFitnessScore - intNewFitnessScore)
+                .difference(intUpdatedFitnessScore - fitnessScoreBefore.getValue())
                 .oldValue(fitnessScoreBefore.getValue())
                 .newValue(intUpdatedFitnessScore)
                 .reason(RewardChangeReason.CONTENT_REVIEWED)
-                .associatedContentIds(getIds(contents))
+                .associatedContentIds(List.of(event.getContentId()))
                 .build();
 
         fitnessScoreBefore.setValue(intUpdatedFitnessScore);
         fitnessScoreBefore.getLog().add(logEntry);
-        System.out.println("Fitness Score Before: " + fitnessScoreBefore.getValue());
-        System.out.println("Fitness Decrease: " + fitnessDecrease);
-        System.out.println("New Fitness Score: " + newFitnessScore);
-        System.out.println("Correctness Before: " + correctnessBefore);
-        System.out.println("Correctness After: " + correctnessAfter);
-        System.out.println("Contents To Repeat: " + contentsToRepeat);
-        System.out.println("Fitness Regen: " + fitnessRegen);
-        System.out.println("Updated Fitness Score: " + updatedFitnessScore);
 
         return fitnessScoreBefore;
     }
+
     private List<UUID> getIds(List<Content> contents) {
-        List<UUID> ids = new ArrayList<>();
-        for (Content content : contents) {
-            ids.add(content.getId());
-        }
-        return ids;
+        return contents.stream().map(Content::getId).toList();
+    }
+
+    private Content getContentOfEvent(List<Content> contents, UserProgressLogEvent event) {
+        return contents.stream()
+                .filter(c -> c.getId().equals(event.getContentId()))
+                .findFirst()
+                .orElseThrow();
     }
 
-    private double calculateFitnessDecrease(List<Content> contents, OffsetDateTime today) {
+    private boolean wasAlreadyLearnedToday(ProgressLogItem lastReview) {
+        return lastReview.getTimestamp().truncatedTo(ChronoUnit.DAYS)
+                .equals(OffsetDateTime.now().truncatedTo(ChronoUnit.DAYS));
+    }
+
+    private double calculateFitnessDecrease(List<Content> contents) {
         double fitnessDecrease = 0.0;
 
         for (Content content : contents) {
-            if (isDueForRepetition(content)) {
+            if (isDueForRepetition(content) && isNotNew(content)) {
```

Comment:
also important: contents that were never learned may not be considered for the fitness